test/configuration_server_test.erl

```diff
@@ -52,11 +52,38 @@ init_test() ->
     meck:expect(filelib, is_regular, fun (_Path) -> true end),
     ?assertEqual(
         {ok, {configuration, filename:absname("config.ini")}},
-        configuration_server:init([])
+        configuration_server:init([no_path])
     ),
     ?assert(meck:validate(filelib)),
     meck:expect(filelib, is_regular, fun (_Path) -> false end),
     ?assertEqual(
         {stop, {no_file, "No config file found"}},
-        configuration_server:init([])
-    ).
+        configuration_server:init([no_path])
+    ),
+    meck:unload(filelib).
+
+init_with_valid_config_flag_test() ->
+    meck:new(filelib, [unstick, passthrough]),
+    meck:expect(filelib, is_regular, fun
+        ("./tep_test.ini") -> true;
+        (Path) -> meck:passthrough([Path])
+    end),
+    ?assertEqual(
+        {ok, {configuration, "./tep_test.ini"}},
```

Comment:
So this is not an issue currently, but just a thought: If our `configuration` record changes (and presumably gains more fields), will that destroy these tests every time we change it? Can we easily test only parts of tuples for equality? In case we can't, maybe we should start thinking about whether that influences our design in any way.

Reply:
Since the configuration is a central part of the application and we might use versioning and hot loading code at some time in the future, I would recommend treating this with care and check every change to it.